File: conformance/tests/bounded-loop/bounded-loop.runar.py
from runar import SmartContract, Bigint, public, assert_

class BoundedLoop(SmartContract):
    expected_sum: Bigint

    def __init__(self, expected_sum: Bigint):
        super().__init__(expected_sum)
        self.expected_sum = expected_sum

    @public
    def verify(self, start: Bigint):
        sum_: Bigint = 0
        for i in range(5):
            sum_ = sum_ + start + i
        assert_(sum_ == self.expected_sum)

File: conformance/tests/bounded-loop/bounded-loop.runar.rb
require 'runar'

class BoundedLoop < Runar::SmartContract
  prop :expected_sum, Bigint

  def initialize(expected_sum)
    super(expected_sum)
    @expected_sum = expected_sum
  end

  runar_public start: Bigint
  def verify(start)
    sum = 0
    for i in 0...5
      sum = sum + start + i
    end
    assert sum == @expected_sum
  end
end
